src/test/suite/editors/activeConnectionCodeLensProvider.test.ts

```diff
@@ -21,34 +21,38 @@ suite('Active Connection CodeLens Provider Test Suite', () => {
     extensionContextStub
   );
   const testStatusView = new StatusView(extensionContextStub);
+  let testConnectionController: ConnectionController;
+  let testCodeLensProvider: ActiveConnectionCodeLensProvider;
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    testConnectionController = new ConnectionController({
+      statusView: testStatusView,
+      storageController: testStorageController,
+      telemetryService: testTelemetryService,
+    });
+    testCodeLensProvider = new ActiveConnectionCodeLensProvider(
+      testConnectionController
+    );
+  });
 
-  suite('the MongoDB playground in JS', () => {
-    const sandbox = sinon.createSandbox();
+  afterEach(() => {
+    sandbox.restore();
+  });
 
+  suite('the MongoDB playground in JS', () => {
     suite('user is not connected', () => {
-      const testConnectionController = new ConnectionController(
-        testStatusView,
-        testStorageController,
-        testTelemetryService
-      );
-      const testCodeLensProvider = new ActiveConnectionCodeLensProvider(
-        testConnectionController
-      );
-      const fakeShowQuickPick = sandbox.fake();
-
       beforeEach(() => {
         testCodeLensProvider.setActiveTextEditor(
           vscode.window.activeTextEditor
         );
+        const fakeShowQuickPick = sandbox.fake();
         sandbox.replace(vscode.window, 'showQuickPick', fakeShowQuickPick);
         const fakeIsPlayground = sandbox.fake.returns(true);
         sandbox.replace(testCodeLensProvider, 'isPlayground', fakeIsPlayground);
       });
 
-      afterEach(() => {
-        sandbox.restore();
```

Comment:
Why don't we want anymore to restore the sandbox after each test?

Reply:
This has just moved up to the top-level `suite()`’s `afterEach()` hook, right?

Reply:
But will it restore stubs for tests inside nested suites? I thought that in this case, it will restore stubs are each suite, not after each test inside nested suites.

Reply:
A higher-level `afterEach()` hook also runs after each test in nested suites, yes